packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -28,27 +27,55 @@ while getopts "r:" option; do
     esac
 done
 
-if [ -z "$resourceGroupName" ]; then
+if [ -z $resourceGroupName ]; then
     exitWithUsageInfo
 fi
 
+# Will return the function name if we deployed before the function app template on this resource group
+functionAppName=$(az group deployment show -g "$resourceGroupName" -n "function-app-template"  --query "properties.parameters.name.value" -o tsv)
+
+# Boolean to track if we should create a new App Registration or there's one already existing
+createNewApp=true
+ 
+# Check if the Function App name is valid and the Function App exists
+if [ ! -z "$functionAppName" ] && (az functionapp show -n "$functionAppName" -g "$resourceGroupName" 1>/dev/null); then
+    # Get the Client (App) ID for the App Registration that's used for this Function App authentication
+    clientId=$(az webapp auth show -n "$functionAppName" -g "$resourceGroupName" --query "clientId" -o tsv)
+    # Check if the App Registration name is valid and the App Registration exists
+    if [ ! -z "$clientId" ] && (az ad app show --id "$clientId" 1>/dev/null); then
+        # If we hit this line, that means we don't need to create a new App Registration
+        createNewApp=false
+    fi
+fi
+
+if ("$createNewApp" = true); then
+    appRegistrationName="allyappregistration_${resourceGroupName}"
+    clientId=$(az ad app create --display-name "$appRegistrationName" --query "appId" -o tsv)
+    echo "Successfully created '$appRegistrationName' App Registration with Client ID '$clientId'"
+else
+    appRegistrationName=$(az ad app show --id "$clientId" --query "displayName" -o tsv)
+    echo "'$appRegistrationName' App Registration with Client ID '$clientId' already exists"
+fi
+
 # Start deployment
 echo "Deploying Function App using ARM template"
 resources=$(az group deployment create \
     --resource-group "$resourceGroupName" \
     --template-file "$templateFilePath" \
+    --parameters '{ "clientId": {"value":"'$clientId'"}}' \
     --query "properties.outputResources[].id" \
     -o tsv)
 
 . "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
 functionAppName=$resourceName
 echo "Successfully deployed Function App '$functionAppName'"
 
-# Add system-assigned managed identity to function app
-
-echo "Adding managed identity to Function App '$functionAppName'"
-principalId=$(az webapp identity assign --name "$functionAppName" --resource-group "$resourceGroupName" --query principalId -o tsv)
-echo "Successfully Added managed identity to Function App '$functionAppName'"
+# Enable continue on error in case the redirect URl was already added before
+set +e
+# add the reply url to the app registration reply urls
+az ad app update --id $clientId --add replyUrls "https://${functionAppName}.azurewebsites.net/.auth/login/aad/callback"
```

Comment:
i dont get any error when reply url is already set.

Reply:
Add this line in a bash script then run it
![image](https://user-images.githubusercontent.com/45858632/64307838-23681a80-cf4c-11e9-8afa-a570560a79dc.png)